### PR TITLE
feat: add @types/relationship.js

### PR DESCRIPTION
**Pull request by whinc**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. package.json
1. types/relationship.js/package.json
1. types/relationship.js/tsconfig.json
1. types/relationship.js/index.d.ts
1. types/relationship.js/relationship.js-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a543afbb36fc8d62c03892e3c954d794c8ad0835
</p>

**Review by navya9singh — APPROVED** [review]

**Comment by whinc** [comment]
Ready to merge

File: package.json [hunk 1]
```diff
@@ -23,8 +23,8 @@
         "test-all": "node --enable-source-maps node_modules/@definitelytyped/dtslint-runner/ --path .",
         "clean": "node scripts/remove-empty.js",
         "clean-node-modules": "node scripts/clean-node-modules.js",
-        "test": "node --enable-source-maps node_modules/@definitelytyped/dtslint/ types",
-        "lint": "node --enable-source-maps node_modules/@definitelytyped/dtslint/ types",
+        "test": "node --enable-source-maps node_modules/@definitelytyped/dtslint/ types/relationship.js",
+        "lint": "node --enable-source-maps node_modules/@definitelytyped/dtslint/ types/relationship.js",
```
Comment (eps1lon) [inline package.json:27]: This command is supposed to be used for testing any package. You can use it as `pnpm run tst relationship.js`. Reverting this specific change in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67793

Comment (whinc) [reply package.json:27]: Got it. Thank you

**Pull request merged**